samples/bluetooth/iso_broadcast_benchmark/src/receiver.c

```diff
@@ -0,0 +1,495 @@
+		      struct net_buf_simple *buf)
+{
+	char le_addr[BT_ADDR_LE_STR_LEN];
+	char name[DEVICE_NAME_LEN];
+
+	if (broadcaster_found) {
+		return;
+	}
+
+	(void)memset(name, 0, sizeof(name));
+
+	bt_data_parse(buf, data_cb, name);
+
+	if (strncmp(DEVICE_NAME, name, strlen(DEVICE_NAME))) {
+		return;
+	}
+
+	bt_addr_le_to_str(info->addr, le_addr, sizeof(le_addr));
+
+	LOG_INF("Found broadcaster with address %s (RSSI %i)",
+		le_addr, info->rssi);
+
+	broadcaster_found = true;
+
+	per_sid = info->sid;
+	per_interval_ms = BT_INTERVAL_TO_MS(info->interval);
+	bt_addr_le_copy(&per_addr, info->addr);
+
+	k_sem_give(&sem_per_adv);
+}
+
+static struct bt_le_scan_cb scan_callbacks = {
+	.recv = scan_recv,
+};
+
+static void sync_cb(struct bt_le_per_adv_sync *sync,
+		    struct bt_le_per_adv_sync_synced_info *info)
+{
+	LOG_INF("Periodic advertisement synced");
+
+	k_sem_give(&sem_per_sync);
+}
+
+static void term_cb(struct bt_le_per_adv_sync *sync,
+		    const struct bt_le_per_adv_sync_term_info *info)
+{
+	LOG_INF("Periodic advertisement sync terminated");
+
+	per_adv_lost = true;
+	k_sem_give(&sem_per_sync_lost);
+}
+
+static void biginfo_cb(struct bt_le_per_adv_sync *sync,
+		       const struct bt_iso_biginfo *biginfo)
+{
+	if (biginfo_received) {
+		return;
+	}
+
+	LOG_INF("BIGinfo received: num_bis %u, nse %u, interval %.2f ms, "
+		"bn %u, pto %u, irc %u, max_pdu %u, sdu_interval %u us, "
+		"max_sdu %u, phy %s, %s framing, %sencrypted",
+		biginfo->num_bis, biginfo->sub_evt_count,
+		BT_INTERVAL_TO_MS((float)biginfo->iso_interval),
+		biginfo->burst_number, biginfo->offset, biginfo->rep_count,
+		biginfo->max_pdu, biginfo->sdu_interval, biginfo->max_sdu,
+		phy2str(biginfo->phy), biginfo->framing ? "with" : "without",
+		biginfo->encryption ? "" : "not ");
+
+	iso_interval_ms = BT_INTERVAL_TO_MS(biginfo->iso_interval);
+	bis_count = MIN(biginfo->num_bis, CONFIG_BT_ISO_MAX_CHAN);
+	biginfo_received = true;
+	k_sem_give(&sem_per_big_info);
+}
+
+static struct bt_le_per_adv_sync_cb sync_callbacks = {
+	.synced = sync_cb,
+	.term = term_cb,
+	.biginfo = biginfo_cb,
+};
+
+static void print_stats(char *name, struct iso_recv_stats *stats)
+{
+	uint32_t total_packets;
+
+	total_packets = stats->iso_recv_count + stats->iso_lost_count;
+
+	LOG_INF("%s: Received %u/%u (%.2f%%) - Total packets lost %u",
+		name, stats->iso_recv_count, total_packets,
+		(float)stats->iso_recv_count * 100 / total_packets,
+		stats->iso_lost_count);
+
+}
+
+static void iso_recv(struct bt_iso_chan *chan,
+		     const struct bt_iso_recv_info *info,
+		     struct net_buf *buf)
+{
+	uint32_t total_packets;
+	static bool stats_latest_arr[1000];
+	static size_t stats_latest_arr_pos;
+
+	/* NOTE: The packets received may be on different BISes */
+
+	if (info->flags == BT_ISO_FLAGS_VALID) {
+		stats_current_sync.iso_recv_count++;
+		stats_overall.iso_recv_count++;
+		stats_latest_arr[stats_latest_arr_pos++] = true;
+	} else {
+		stats_current_sync.iso_lost_count++;
```

Comment:
Yet another question ;-)
Will there always be a iso_recv callback, or can a packet be so lost that there is not even a callback?  (In which case, the stats are not completely reliable.)

Reply:
We may discard packets, yes, either if they are wrongly formatted by the controller or if they are too large for our buffers in the host.

Reply:
Will this lead to an incorrect "lost" count?

Reply:
Depends on how you look at it I suppose. From a Link Layer/host perspective they are not lost, but merely badly formatted/too large. The application won't be notified about those, but that's something outside the scope of this PR, and possibly something we can update.